src/System.Private.ServiceModel/src/Internals/System/Runtime/TimeoutHelper.cs

```diff
@@ -300,7 +300,7 @@ public static Task<CancellationToken> FromTimeoutAsync(int millisecondsTimeout)
 
             if (!s_tokenCache.TryGetValue(targetTime, out tokenTask))
             {
-                var tcs = new TaskCompletionSource<CancellationToken>();
+                var tcs = new TaskCompletionSource<CancellationToken>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Can we add instrumentation for the helper method so that whoever awaits on tcs.Task would check for its OperationalContext.Current before and after the await and DebugBreaks if they are not equal? Essentially I want to ensure that there are no other cases when preempting somebody's else threads gets their OC.